Add click-able local storage items

icyflame opened the pull request:
- local storage is populated using the value of the query whenever the user clicked on a link
- every item in the local storage is a clickable item

The diff is slightly long because I have moved all the CSS and JS to separate files to make it easier to understand. Diffs for everything after the first commit should be easily understandable.

Earlier discussion was here: #12 

Live demo: http://icyflame.github.io/mfqp/

Screenshots:

![image](https://cloud.githubusercontent.com/assets/3668034/25064202/27436e22-2213-11e7-99fc-90668f1bd934.png)
![image](https://cloud.githubusercontent.com/assets/3668034/25064206/2e7b18c0-2213-11e7-8cf2-45b5a81162f2.png)

Fixes #7

icyflame committed:
move all css and javascript to separate files

icyflame committed:
add both local storage and pre-filling of queries from URL

icyflame committed:
indent index.html properly

icyflame committed:
make the local storage items click-able

- clicking on the item will fill the query box and then execute a search

athityakumar reviewed (commented):
Works. :+1: 

From a UX point of view, wouldn't it be better if the last searched query was on the top of the "Previous searches" list?

icyflame commented:
@athityakumar Yeah, it would. Thanks for pointing it out! I will reverse chronological order is the one that makes sense.

icyflame commented:
@vivekiitkgp @amrav Please review this PR. I will merge this in a few hours, after a little bit more testing, I feel confident about it though.

@athityakumar I have noted your suggestion, unfortunately, I can't work on them right now. I am adding an issue for now, and I will work on them later. I think we should push this out this exam session!

athityakumar commented on client.js:
```diff
@@ -0,0 +1,113 @@
+      // this is a result link
+      var query_value = $("#query").val().trim();
+      if(localStorage.getItem("searched")) {
+        localStorage.setItem("searched", localStorage.getItem("searched") + "," + query_value);
```
Line 32 of `client.js`

Won't `localStorage.setItem("searched",  query_value + "," + localStorage.getItem("searched"))` do the job of chronological order? 

Ping @icyflame

icyflame replied:
Only on the next reload of the page.

When the page loads, I get all the items from the LocalStorage and add an appropriate `<ul></ul>` with `<li>` elements. And then I simply append `<li>` elements to this list.

athityakumar replied:
Hmm, but the recent search does get appended to the list when a paper link is clicked right? Anyway, I think it'd be better if we can get started on the chronological order before merging this PR.

athityakumar commented:
@icyflame - We should also probably add an issue for the mobile responsiveness, which we can add later. :smile:

icyflame commented:
@athityakumar I don't understand, do you think there's something wrong with the mobile view?

athityakumar commented:
@icyflame - Nothing wrong, but it just doesn't look that awesome to me on mobile screen resolution.

![image](https://cloud.githubusercontent.com/assets/17109060/25237275/cec29acc-2607-11e7-89f3-2936087ab327.png)

athityakumar reviewed (approved):
@icyflame - Thanks for making local storage work 🎉 . I think it's high time we merge this. What say?

icyflame commented:
@athityakumar mobile preview still isn't good. that needs to be fixed. if you have time, can you get to it?

athityakumar commented on index.html:
```diff
@@ -196,5 +70,8 @@
-    </body>
+      </div>
+    </div>
+    <script src="local-storage.js"></script>
```
There's no local-storage.js file being used. I'll fix this along with UI part. Sending a PR in an hour.

icyflame replied:
Great, rebase onto the latest master. Thanks! :+1:

athityakumar committed:
Adds some JS enhancements

Unique search results with recent search on top

athityakumar commented:
@icyflame - Can you review https://github.com/icyflame/mfqp/pull/2 please? After that is merged, we can merge this too. 😄

icyflame commented:
Hey @athityakumar, thanks for pointing me there. I will review and merge that. This is WAY over due! :smile:

icyflame committed:
do not hide div if list is empty

icyflame committed:
print debug log only in debug mode, check for query only if it exists

- this ensures no data is printed in the console
- the undefined check ensures that there are no errors in the console

icyflame committed:
ensure that searches aren't repeated simply because of case differences

icyflame commented:
@amrav This patch is finally done! Can you please review this?

A live demo is available at http://icyflame.github.io/mfqp

## Preview on a computer

![image](https://user-images.githubusercontent.com/3668034/32940563-0365c1cc-cba9-11e7-840a-b5bc970fd51b.png)

## Preview on a phone

![screenshot_20171117-145921](https://user-images.githubusercontent.com/3668034/32940602-1d98b68a-cba9-11e7-9c8a-af9d3284f07c.png)

icyflame commented:
Hey @amrav, slight nudge for a review. :slightly_smiling_face:

amrav commented:
@icyflame hey sorry for the delayed review! Code LGTM, but I'm not a fan of the two column layout. Google solves this by showing recent searches in the search bar itself, on focus, which is a neat solution. Another possibility is to have a two row layout, with a horizontal (wrapped) list of recent searches just beneath the search bar.

If you feel strongly about the current layout, I'm happy to merge as-is 👍

thealphadollar commented:
@icyflame  PR has been stagnant for a long time.

ghostwriternr commented:
I'm with Vikrant on this too. This two column layout seems a bit weird and pushes focus to the side. I like the simplicity that we have with MFQP right now. If something like the way Google does it can be added for these recent searches though, that would be pretty neat!

icyflame committed:
change from two column to displaying recent searches after results

icyflame commented:
@ghostwriternr @amrav noted. I have fixed the layout to show the recent searches after the results. This doesn't disturb the present UI. Since the discussion has gone on for long enough and this change is pretty small (UI wise), I will merge this in tomorrow unless someone raises some problems with the code. (I have re-indented the code so the diff might be a bit too big)

[Final Demo](https://www.siddharthkannan.in/mfqp/)

icyflame commented:
Force pushed to rebase to latest gh-pages. The links weren't working before (old mfqp json file).

The diff isn't that big actually. It's easy to read too, so please do if you can.

Merging tomorrow morning.
